File: 2025/day-11-reactor/src/device.rs
#[derive(Clone, PartialEq)]
pub struct Device {
    pub input: String,
    pub output: Vec<String>,
}

impl Device {
    pub fn new(line: &str) -> Device {
        let split = line.split(":").collect::<Vec<&str>>();

        Device {
            input: split[0].to_string(),
            output: Self::parse_output(split[1]),
        }
    }

    pub fn is_out(&self) -> bool {
        if self.output.len() == 1 && self.output[0] == "out" {
            return true;
        }
        false
    }

    pub fn parse_output(raw_output: &str) -> Vec<String> {
        raw_output
            .split(" ")
            .filter(|s| !s.is_empty())
            .map(|s| s.trim().to_string())
            .collect()
    }

    pub fn print(&self) {
        println!("Input: {}, Output: {:?}", self.input, self.output);
    }
}

File: 2025/day-11-reactor-pt2/Device.cs
public class Device {
  public String input;
  public String[] outputs;

  public Device(String line) {
    String[] split = line.Split(":");
    String[] outputs = split[1].Split(" ").Select(s => s.Trim()).Where(s => !string.IsNullOrWhiteSpace(s)).ToArray();

    this.input = split[0].Trim();
    this.outputs = outputs;
  }

  public bool IsOut() => outputs.Length == 1 && outputs[0] == "out";

  public bool IsStart() => input == "svr";

  public bool IsDac() => input == "dac";

  public bool IsFFT() => input == "fft";

  public override string ToString() => $"Device: input: '{this.input}', outputs: [{string.Join(", ", this.outputs)}]";
}
